app/src/main/java/com/example/userinfo/Validation.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.userinfo
+
+import android.widget.EditText
+
+class Validation {
+
+    fun checkAllFields(
+        fields: ArrayList<EditText>
```

Comment:
If we are passing list of EditText, we made this class dependant on Android View System.
You can only pass the values here and do the validation

If it does not depend on explicit views, we can re-use it elsewhere and write local tests only( instead of instrument tests)